pages/docs/tutorials/native/dynamic-libraries.md

```diff
@@ -0,0 +1,346 @@
+---
+type: tutorial
+layout: tutorial
+title:  "Kotlin/Native as a Dynamic Library"
+description: "Compiling Kotlin/Native code to a dynamic library"
+authors: Eugene Petrenko
+date: 2018-08-05
+showAuthorInfo: false
+issue: EVAN-5371
+---
+
+In the tutorial, we see how to use a Kotlin/Native code from
+an existing native applications or libraries. For that we 
+compile our Kotlin code into a dynamic library, `.so`, `.dylib` and `.dll`.
+
+Targeting Apple platform? You may want to see the 
+[Kotlin/Native as an Apple Framework](apple-framework.html)
+tutorial.
+
+In this tutorial, we will:
+ - [Compile a Kotlin code to a dynamic library](#creating-a-kotlin-library)
+ - [Examine generated C headers](#generated-headers-file)
+ - [Use the Kotlin dynamic library from C](#using-generated-headers-from-c)
+ - Compile and run the example [on Linux and Mac](#compiling-and-running-the-example-on-linux-and-macos)
+   and [on Windows](#compiling-and-running-the-example-on-windows)
+  
+## Creating a Kotlin Library
+
+Kotlin/Native compiler is able to produce a dynamic
+library out of the Kotlin code we have.
+A produced dynamic library often comes with a header file, an `.h` file,
+which we use to call compiled code from C
+
+The best way to understand the techniques is to try those techniques. 
+Let's create a tiny Kotlin library first and use it from C program 
+than. 
+
+Let's create a library file in Kotlin and save it as `lib.kt`:
+```kotlin
+package demo
+
+class DemoClass {
+  fun foo() : Long = 42
+}
+
+fun ints(b: Byte, s: Short, i: Int, l:Long) { }
+fun floats(f: Float, d: Double) { }
+
+fun strings(str: String) : String {
+  return "That is '$str' from C"
+}
+
+val globalString = "A global String"
+```
+
+We need to have a Kotlin/Native compiler on our machines. 
+You may have a look at the
+[A Basic Kotlin/Native Application](basic-kotlin-native-app.html#obtaining-the-compiler)
+tutorial for more information on performing this step.
+Let's assume we have a console, where `kotlinc` command is available. 
+
+Now we call the following command to compile the code into a dynamic library:
+```bash
+kotlinc lib.kt -produce dynamic -output demo
+```
+
+The `kotlinc` generates the following files, depending on the OS, 
+where you run the compiler:
+- macOS: `demo_api.h` and `libdemo.dylib`
+- Linux: `demo_api.h` and `libdemo.so`
+- Windows: `demo_api.h` and `demo.dll`
```

Comment:
On Windows .def file is produced as well. Seems due to small bug it was produced inside Windows temporary files, but it will be fixed with https://github.com/JetBrains/kotlin-native/pull/1859.

Reply:
fixed in the dynamic libraries branch and #1084